app/src/main/java/com/nmakademija/nmaakademija/adapter/UserListAdapter.java

```diff
@@ -1,47 +1,103 @@
 package com.nmakademija.nmaakademija.adapter;
 
 import android.content.Context;
-import android.support.v4.app.Fragment;
-import android.support.v4.app.FragmentManager;
-import android.support.v4.app.FragmentStatePagerAdapter;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ArrayAdapter;
+import android.widget.Filter;
+import android.widget.Filterable;
+import android.widget.ImageView;
+import android.widget.TextView;
 
+import com.bumptech.glide.Glide;
 import com.nmakademija.nmaakademija.R;
+import com.nmakademija.nmaakademija.entity.Section;
 import com.nmakademija.nmaakademija.entity.User;
-import com.nmakademija.nmaakademija.fragment.UserListFragment;
 
 import java.util.ArrayList;
+import java.util.List;
 
-public class UserListAdapter extends FragmentStatePagerAdapter {
+public class UserListAdapter extends ArrayAdapter<User> implements Filterable {
     private Context context;
     private ArrayList<User> users;
+    private ArrayList<User> allUsers;
+    private List<Section> sections;
 
-    public UserListAdapter(Context c, ArrayList<User> u, FragmentManager fm) {
-        super(fm);
-        context = c;
-        users = u;
+    public UserListAdapter(Context context, ArrayList<User> users, List<Section> sections) {
```

Comment:
Change `users` to `List<User>`. In this case you won't need a cast to pass argument